internal/backend/remote-state/s3/backend.go

```diff
@@ -746,16 +752,12 @@ func (b *Backend) Configure(obj cty.Value) tfdiags.Diagnostics {
 		StsEndpoint:             customEndpoints["sts"].String(obj),
 		StsRegion:               stringAttr(obj, "sts_region"),
 		Token:                   stringAttr(obj, "token"),
-
-		// Note: we don't need to read env variables explicitly because they are read implicitly by aws-sdk-base-go:
-		// see: https://github.com/hashicorp/aws-sdk-go-base/blob/v2.0.0-beta.41/internal/config/config.go#L133
-		// which relies on: https://cs.opensource.google/go/x/net/+/refs/tags/v0.18.0:http/httpproxy/proxy.go;l=89-96
-		HTTPProxy:            aws.String(stringAttrDefaultEnvVar(obj, "http_proxy", "HTTP_PROXY")),
-		HTTPSProxy:           aws.String(stringAttrDefaultEnvVar(obj, "https_proxy", "HTTPS_PROXY")),
-		NoProxy:              stringAttrDefaultEnvVar(obj, "no_proxy", "NO_PROXY"),
-		Insecure:             boolAttr(obj, "insecure"),
-		UseDualStackEndpoint: boolAttr(obj, "use_dualstack_endpoint"),
-		UseFIPSEndpoint:      boolAttr(obj, "use_fips_endpoint"),
+		HTTPProxy:               aws.String(stringAttrDefaultEnvVar(obj, "http_proxy", "http_proxy", "HTTP_PROXY")),
+		HTTPSProxy:              aws.String(stringAttrDefaultEnvVar(obj, "https_proxy", "https_proxy", "HTTPS_PROXY")),
+		NoProxy:                 stringAttrDefaultEnvVar(obj, "no_proxy", "no_proxy", "NO_PROXY"),
+		Insecure:                boolAttr(obj, "insecure"),
+		UseDualStackEndpoint:    boolAttr(obj, "use_dualstack_endpoint"),
+		UseFIPSEndpoint:         boolAttr(obj, "use_fips_endpoint"),
```

Comment:
Hi - I'm the person who originally reported this issue in #1594.

I disagree that this is the correct fix.

Ultimately, the bug was caused by re-implementing logic that a lower library could already perform, and re-implementing it incorrectly.

This changes doubles down on the reimplementation... and I can see diverges from it in at least one way, still.

Even if this code was changed to fully match the behaviour of the underlying library, it would still be needless duplication, and a hypothetical future bug if the underlying library was ever updated.

Please see my comment from 16th May on the original issue: https://github.com/opentofu/opentofu/issues/1594#issuecomment-2116200078 in which I suggested how this might be fixed instead.

Thanks.

Reply:
I can test this tomorrow.

Reply:
If we remove the explicit setting of these attributes, then we should not delete the comment in 750-752, and maybe even clarify it a little more and mention which environment variables we are implicitly getting. We can move this comment to the added logic in 783-792, it might make more sense there.

Reply:
`tofu` from this branch successfully completes init/plan in my environment where release versions fail.